### PR TITLE
PATCH: Adds index mismatch interrupt

### PR DESCRIPTION
**Pull request by coachchucksol**
This PR will add in a service routine when a validator is removed. This functionality has been moved from epoch maintenance to its own routine to cover the case of validators being able to be removed on the same epoch if they are delinquent.

### CODE REVIEW

**Comment by coachchucksol** [comment]
Problem:

When running the interrupt, there is currently no way of knowing on-chain which validator needs to be removed. Maybe we can pass in some account that says its been removed?

**Comment by coachchucksol** [comment]
We can avoid adding in extra tracking indicies by looking at the validator list from smallest `marked_for_removal` to highest and check the corresponding state of the validator.

File: programs/steward/src/instructions/idle.rs [hunk 1]
```diff
@@ -14,6 +17,16 @@ pub struct Idle<'info> {
...
+
+    /// CHECK: Correct account guaranteed if address is correct
+    #[account(address = deserialize_stake_pool(&stake_pool)?.validator_list)]
+    pub validator_list: AccountInfo<'info>,
+
+    /// CHECK: Correct account guaranteed if address is correct
+    #[account(
+        address = get_stake_pool_address(&config)?
+    )]
+    pub stake_pool: AccountInfo<'info>,
```
Comment (ebatsell) [inline programs/steward/src/instructions/idle.rs:29]: we added the validator list address to the config account so this isn't necessary

Comment (ebatsell) [reply programs/steward/src/instructions/idle.rs:29]: a few other instructions where we can remove stake_pool as well

**Pull request closed without merging**